src/components/MAPI/organizations/utils.ts

```diff
@@ -34,3 +36,69 @@ export function extractErrorMessage(
 
   return message;
 }
+
+export enum OrganizationNameStatusMessage {
+  TooShort = `Must be at least 4 characters long`,
+  StartAndEndWithAlphaNumeric = 'Must start and end with an alphanumeric character (a-z, 0-9)',
+  CharacterSet = 'Must contain only a-z, 0-9, and hyphens and underscores',
+  TooLong = 'Must not be longer than 249 characters',
+  Ok = '',
+}
+
+const minLength = 4;
+const maxLength = 249;
+
+const startAndEndWithAlphanumericRegExp = /^[a-z0-9].*[a-z0-9]$/;
+const characterSetRegExp = /^[a-z0-9\-\_]+$/;
```

Comment:
Extracted these out of the function, since they don't need to be compiled every time they're used.